comp/dogstatsd/server/parse.go

```diff
@@ -238,8 +246,25 @@ func (p *parser) parseFloat64List(rawFloats []byte) ([]float64, error) {
 }
 
 // extractContainerID parses the value of the container ID field.
+// If the field is prefixed by `in-`, it corresponds to the cgroup controller's inode of the source
+// and is used for ContainerID resolution.
 func (p *parser) extractContainerID(rawContainerIDField []byte) []byte {
-	return rawContainerIDField[len(containerIDFieldPrefix):]
+	containerIDField := rawContainerIDField[len(containerIDFieldPrefix):]
+
+	if bytes.HasPrefix(containerIDField[:len(containerIDFieldInodePrefix)], containerIDFieldInodePrefix) {
+		inodeField, err := strconv.ParseUint(string(containerIDField[len(containerIDFieldPrefix)+1:]), 10, 64)
+		if err != nil {
+			return nil
+		}
+
+		containerID, err := provider.GetProvider().GetMetaCollector().GetContainerIDForInode(inodeField, readerCacheExpiration)
+		if err != nil {
+			return nil
```

Comment:
Ditto.

Confidence check: is it possible that an inode transmitted by an app is invisible/unaccessible for the Agent?

Reply:
Yes, if the host's cgroups aren't mounted. Also, cgroup<>inode mapping is refreshed every 2s.

Reply:
Got it, I've added a debug log.
In that case `GetContainerIDForInode()` will return `("", nil)`, and the tag will simply be dropped.

```
2023-12-15 09:46:52 UTC | CORE | DEBUG | (comp/dogstatsd/server/parse.go:264 in extractContainerID) | Failed to get container ID, got containerID not found from inode 0, err: %!w(<nil>)
```